ddtrace/settings/profiling.py

```diff
@@ -128,6 +134,34 @@ def _check_for_injected():
     return _profiling_injected
 
 
+def _update_git_metadata_tags(tags):
+    """
+    Update profiler tags with git metadata
+    """
+    # clean tags, because values will be combined and inserted back in the same way as for tracer
+    gitmetadata.clean_tags(tags)
+    repository_url, commit_sha, main_package = gitmetadata.get_git_tags()
+    if repository_url:
+        tags[REPOSITORY_URL] = repository_url
+    if commit_sha:
+        tags[COMMIT_SHA] = commit_sha
+    if main_package:
+        tags[MAIN_PACKAGE] = main_package
+    return tags
+
+
+def _enrich_tags(tags) -> t.Dict[str, str]:
+    tags = {
+        k: compat.ensure_text(v, "utf-8")
+        for k, v in itertools.chain(
+            _update_git_metadata_tags(parse_tags_str(os.environ.get("DD_TAGS"))).items(),
```

Comment:
I think this is good.  In a later PR, it would be cool if one of us would consolidate some of our common profiler configuration using the tracer config (it has its own logic for tags/service/hostname/etc extraction, and we should just use it).  I don't think that should go in this PR, though.

Reply:
Agreed. There are quite a few configs that `Profiler` directly reads from `os.environ` instead of tracer config: `DD_SERVICE`, `DD_ENV`, `DD_VERSION` and `DD_API_KEY`.

https://github.com/DataDog/dd-trace-py/blob/4d1e94656daa1c6138d0ace9fe09112d4313dda3/ddtrace/profiling/profiler.py#L128-L133